### PR TITLE
usagestats: Reinstate argument field in BigQuery events, add allowlist for events with public arguments

### PR DESCRIPTION
**Pull request by attfarhan**
Addresses [RFC 437](https://docs.google.com/document/d/1_tYPVYRW-pVhDdJRcF7umcV-2tQ3GCJz6DHBrq5oKb8/edit#). 

We currently do not send the `argument` column to BigQuery at all. This was to avoid potentially sending private data from event logs on Cloud into our analytics systems. However, we do need the non-private data from this column. This PR adds an allowlist of event names that don't log any private data, and sends arguments along with these events to BigQuery.

### CODE REVIEW

**Review by indradhanush — COMMENTED** [review]
This will likely require a schema change in BigQuery. Leaving this link for reference: https://docs.sourcegraph.com/dev/background-information/adding_ping_data#changing-the-bigquery-schema.

File: internal/usagestats/event_handlers_test.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	if diff := cmp.Diff(want, got); diff != "" {
+		t.Fatal(diff)
+	}
+
+}
+func TestCreateBigQueryEvent_PrivateEvent(t *testing.T) {
+	testUrl := "https://sourcegraph.com/test"
+	now := time.Date(2021, 1, 28, 0, 0, 0, 0, time.UTC)
+	mockTimeNow(now)
+	privateArgumentField := "\"code_search\": { \"query_data\": { \"query\": \"test\", \"combined\": \"something private\", \"empty\": \"false\"}"
```
Comment (unknwon) [inline internal/usagestats/event_handlers_test.go:56]: nit: Use backticks to surround this string would make it a lot human-readable.

**Review by unknwon — APPROVED** [review]
Left one comment for code simplification, and approve to unblock.

**Comment by attfarhan** [comment]
Closing, merged https://github.com/sourcegraph/sourcegraph/pull/23627 as a better solution.

**Pull request closed without merging**